spec.html

```diff
@@ -29821,13 +29821,18 @@ <h1>Number.prototype.toPrecision ( _precision_ )</h1>
             1. Let _m_ be the String value consisting of _p_ occurrences of the code unit 0x0030 (DIGIT ZERO).
             1. Let _e_ be 0.
           1. Else,
-            1. Let _e_ and _n_ be integers such that 10<sup>_p_ - 1</sup> &le; _n_ &lt; 10<sup>_p_</sup> and for which _n_ &times; 10<sup>_e_ - _p_ + 1</sup> - _x_ is as close to zero as possible. If there are two such sets of _e_ and _n_, pick the _e_ and _n_ for which _n_ &times; 10<sup>_e_ - _p_ + 1</sup> is larger.
+            1. Let _e_ be floor(the base 10 logarithm of _x_).
+            1. Let _n_ be an integer such that 10<sup>_p_ - 1</sup> &le; _n_ &le; 10<sup>_p_</sup> and for which (_n_ / 10<sup>_p_ - 1</sup>) &times; 10<sup>e</sup> - _x_ is as close to zero as possible. If there are two such _n_, pick the larger _n_. Note that 1 &le; _n_ / 10<sup>_p_ - 1</sup> &le; 10.
+            1. If _n_ = 10<sup>_p_</sup>, then
+              1. Set _e_ to _e_ + 1.
+              1. Set _n_ to 10<sup>_p_ - 1</sup>.
             1. Let _m_ be the String value consisting of the digits of the decimal representation of _n_ (in order, with no leading zeroes).
+            1. Assert: the length of _m_ is _p_.
             1. If _e_ &lt; -6 or _e_ &ge; _p_, then
               1. Assert: _e_ &ne; 0.
               1. If _p_ &ne; 1, then
-                1. Let _a_ be the first code unit of _m_.
-                1. Let _b_ be the other _p_ - 1 code units of _m_.
+                1. Let _a_ be the code unit at index 0 within _m_.
+                1. Let _b_ be the substring of _m_ from 1.
                 1. Set _m_ to the string-concatenation of _a_, *"."*, and _b_.
               1. If _e_ &gt; 0, then
                 1. Let _c_ be the code unit 0x002B (PLUS SIGN).
```

Comment:
I'm also thinking about improving the names of these aliases and replacing the code point references with inline literals.
```suggestion
                1. Let _sign_ be *"+"*.
```